datastore/longhorn.go

```diff
@@ -99,15 +99,33 @@ func (s *DataStore) filterCustomizedDefaultSettings(customizedDefaultSettings ma
 	availableCustomizedDefaultSettings := make(map[string]string)
 
 	for name, value := range customizedDefaultSettings {
-		if err := s.ValidateSetting(string(name), value); err == nil {
-			availableCustomizedDefaultSettings[name] = value
-		} else {
-			logrus.WithError(err).Errorf("invalid customized default setting %v with value %v, will continue applying other customized settings", name, value)
+		if s.isSettingValueChanged(types.SettingName(name), value) {
+			if err := s.ValidateSetting(string(name), value); err == nil {
+				availableCustomizedDefaultSettings[name] = value
+			} else {
+				logrus.WithError(err).Warnf("invalid customized default setting %v with value %v, will continue applying other customized settings", name, value)
+			}
 		}
```

Comment:
Nit: suggest readjusting the if statements. IMO fewer layers can improve readability, ex:
```
	for name, value := range customizedDefaultSettings {
		if !s.isSettingValueChanged(types.SettingName(name), value) {
			continue
		}

		if err := s.ValidateSetting(string(name), value); err != nil {
			logrus.WithError(err).Warnf("invalid customized default setting %v with value %v, will continue applying other customized settings", name, value)
			continue
		}

		availableCustomizedDefaultSettings[name] = value
	}
```

Reply:
Thanks for suggestion. I will fix it to improve readability.

Reply:
Done to fix it.

https://github.com/longhorn/longhorn-manager/blob/9c76065dbfc0e069824afaafc187b982c73bf79e/datastore/longhorn.go#L101-L112